src/why3prop.ml

```diff
@@ -0,0 +1,127 @@
+    with NoGoalTranslation ->
+        Console.fatal pos "The term [%a] is not of the form [P _]"
+        Print.pp g
+
+(** [translate_context cfg (l_constants, l_hypothesis) (hyp_name, (_, hyp))]
+    translate the context [hyp] with the label [hyp_name] and add it in
+    [l_hypothesis] with the why3 constants [l_constants]. *)
+and translate_context :
+    prop_config ->
+    cnst_table * ctxt_labels ->
+    string * (tvar * tbox) ->
+    cnst_table * ctxt_labels =
+    fun cfg (l_constants, l_hypothesis) (hyp_name, (_, hyp)) ->
+    try
+        let (new_why3_l, hyp') = t_goal cfg l_constants (Bindlib.unbox hyp) in
+            (new_why3_l, (hyp_name, hyp')::l_hypothesis)
+    with NoGoalTranslation ->
+            (l_constants, l_hypothesis)
+
+(** [t_goal cfg constants_table trm] translate the lambdapi term [trm] to Why3
+    term using the configuration [cfg] and the list of Why3 constants in
+    [constants_table]. *)
+and t_goal : prop_config -> cnst_table -> term ->
+    cnst_table * Why3.Term.term =
+    fun cfg constants_table trm ->
+    match Basics.get_args trm with
+    | (symbol, [t]) when Basics.is_symb cfg.symb_P symbol ->
+        t_prop cfg constants_table [] t
+    | _                                                         ->
+        raise NoGoalTranslation
+
+(** [t_prop cfg constants_table ctxt p] translate the term [p] into Why3 terms
+    with a context [ctxt] and a config [cfg]. *)
+and t_prop :
+    prop_config -> cnst_table -> Ctxt.t -> term ->
+    cnst_table * Why3.Term.term =
+    fun cfg constants_table ctxt p ->
+    match Basics.get_args p with
+    | symbol, [t1; t2] when Basics.is_symb cfg.symb_and symbol  ->
+        let (constants_table, t1) = t_prop cfg constants_table ctxt t1 in
+        let (constants_table, t2) = t_prop cfg constants_table ctxt t2 in
+        constants_table, Why3.Term.t_and t1 t2
+    | symbol, [t1; t2] when Basics.is_symb cfg.symb_or symbol  ->
+        let (constants_table, t1) = t_prop cfg constants_table ctxt t1 in
+        let (constants_table, t2) = t_prop cfg constants_table ctxt t2 in
+        constants_table, Why3.Term.t_or t1 t2
+    | symbol, [t1; t2] when Basics.is_symb cfg.symb_imp symbol  ->
+        let (constants_table, t1) = t_prop cfg constants_table ctxt t1 in
+        let (constants_table, t2) = t_prop cfg constants_table ctxt t2 in
+        constants_table, Why3.Term.t_implies t1 t2
+    | symbol, [t] when Basics.is_symb cfg.symb_not symbol  ->
+        let (constants_table, t) = t_prop cfg constants_table ctxt t in
+        constants_table, Why3.Term.t_not t
+    | _                                                     ->
+        (* if the term [p] is in the list [constants_table] *)
+        if List.exists (fun (lp_t, _) -> Basics.eq lp_t p) constants_table
+        then
+            (* then find it and return it *)
+            let lp_eq = fun (lp_t, _) -> Basics.eq lp_t p in
+            let (_, ct) = List.find lp_eq constants_table in
```

Comment:
You should not call List.exists and then List.find but call List.find directly catch the exception Not_found.

Reply:
Fixed in 5afd69dcf6e69ca055a3b196fbd5e2f36869081d